src/main/java/org/sunbird/workflow/service/impl/NotificationServiceImpl.java

```diff
@@ -169,11 +174,67 @@ public void sendEmailNotification(WfRequest wfRequest) {
+			logger.info("Notification workflow status model, {}", mapper.writeValueAsString(wfStatus));
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+		if (!ObjectUtils.isEmpty(wfStatus.getNotificationEnable()) && wfStatus.getNotificationEnable()) {
+			logger.info("Enter in the notification block");
+			List<Map<String, Object>> mdoAdminList = userProfileWfService.getMdoAdminDetails(wfRequest.getRootOrgId());
+			Map<String, Object> params = new HashMap<>();
+			NotificationRequest request = new NotificationRequest();
+			request.setDeliveryType("message");
+			List<String> mdoMailList = mdoAdminList.stream().map(mdoAdmin -> (String) mdoAdmin.get("email")).collect(Collectors.toList());
+			if (!CollectionUtils.isEmpty(mdoMailList)) {
+				request.setIds(mdoMailList);
+				request.setMode("email");
+				Template template = new Template();
+				template.setId(EMAILTEMPLATE);
+				Optional<HashMap<String, Object>> updatedFieldValue = wfRequest.getUpdateFieldValues().stream().findFirst();
+				HashMap<String, Object> usersObj = userProfileWfService.getUsersResult(Collections.singleton(wfRequest.getUserId()));
+				Map<String, Object> recipientInfo = (Map<String, Object>)usersObj.get(wfStatusEntity.getUserId());
+				if (updatedFieldValue.isPresent()) {
+					params.put("body", constructMdoEmailBody((String) recipientInfo.get(Constants.FIRST_NAME), updatedFieldValue, wfRequest.getApplicationId()));
+				}
+				params.put("orgImageUrl", null);
+				template.setParams(params);
+				Config config = new Config();
+				config.setSubject(MDO_MAIL_SUBJECT);
+				config.setSender(configuration.getSenderMail());
+				Map<String, Object> req = new HashMap<>();
+				request.setTemplate(template);
+				request.setConfig(config);
+				Map<String, List<NotificationRequest>> notificationMap = new HashMap<>();
+				notificationMap.put("notifications", Arrays.asList(request));
+				req.put("request", notificationMap);
+				sendNotification(req);
+			}
+		}
+	}
 
-	/**
-	 * Post to the Notification service
-	 * @param request
-	 */
+	public String constructMdoEmailBody(String userName, Optional<HashMap<String, Object>> updatedFieldValue, String applicationId) {
+		HashMap<String, Object> toValue = (HashMap<String, Object>) updatedFieldValue.get().get(TO_VALUE_CONST);
+		List<String> fieldNames = toValue.keySet().stream().collect(Collectors.toList());
+		String requestString = IntStream.range(0, fieldNames.size())
+				.mapToObj(i -> "&nbsp;&nbsp;" + (i + 1) + ". " + fieldNames.get(i).toUpperCase())
+				.collect(Collectors.joining("<br>"));
+		requestString = "Dear Admin, <br><br> &nbsp; You have received the below requests for approval from " + userName + "<br>" + requestString;
+		String approvalUrl = configuration.getMdoBaseUrl() + applicationId + Constants.TO_APPROVE_CONST;
+		String body = requestString + "<br><br> Please Click <a href='" + approvalUrl + "' style='color: blue;'>here</a> to view the requests.";
+		return body;
+	}
+
```

Comment:
why are we constructing the request body for email ?
you suppose to use email_template for this right ?? if any format change is requested, we can just work on the html format alone... but this would require a code change each time.

Reply:
yes created a vm template file now please help me how to implement this template